src/ray/gcs/gcs_server/gcs_kv_manager.cc

```diff
@@ -33,6 +33,25 @@ void GcsInternalKVManager::HandleInternalKVGet(
       }));
 }
 
+void GcsInternalKVManager::HandleInternalKVMultiGet(
+    const rpc::InternalKVMultiGetRequest &request, rpc::InternalKVMultiGetReply *reply,
+    rpc::SendReplyCallback send_reply_callback) {
+  auto cnt = std::make_shared<int>(request.keys_size());
+  for (int i = 0; i < request.keys_size(); ++i) {
+    std::vector<std::string> cmd = {"HGET", request.keys(i), "value"};
+    RAY_CHECK_OK(redis_client_->GetPrimaryContext()->RunArgvAsync(
```

Comment:
Is this thread safe? Especially `redis_client_->GetPrimaryContext()->RunArgvAsync(`?

Reply:
it is